rebasehelper/logger.py

```diff
@@ -64,38 +66,85 @@ def log(message, *args, **kwargs):
+            CustomLogger.TRACE: {'fg': 'red', 'bg': 'default', 'style': None},
+            CustomLogger.VERBOSE: {'fg': 'brightblack', 'bg': 'default', 'style': None},
+            logging.INFO: {'fg': 'default', 'bg': 'default', 'style': None},
+            CustomLogger.SUCCESS: {'fg': 'green', 'bg': 'default', 'style': None},
+            CustomLogger.HEADING: {'fg': 'yellow', 'bg': 'default', 'style': None},
+            CustomLogger.IMPORTANT: {'fg': 'red', 'bg': 'default', 'style': None},
+            logging.WARNING: {'fg': 'yellow', 'bg': 'default', 'style': None},
+            logging.ERROR: {'fg': 'red', 'bg': 'default', 'style': 'bold'},
+            logging.CRITICAL: {'fg': 'white', 'bg': 'red', 'style': 'bold'},
+        },
+        'light': {
+            logging.DEBUG: {'fg': 'brightblack', 'bg': 'default', 'style': None},
+            CustomLogger.TRACE: {'fg': 'red', 'bg': 'default', 'style': None},
+            CustomLogger.VERBOSE: {'fg': 'brightblack', 'bg': 'default', 'style': None},
+            logging.INFO: {'fg': 'default', 'bg': 'default', 'style': None},
+            CustomLogger.SUCCESS: {'fg': 'green', 'bg': 'default', 'style': None},
+            CustomLogger.HEADING: {'fg': 'blue', 'bg': 'default', 'style': None},
+            CustomLogger.IMPORTANT: {'fg': 'red', 'bg': 'default', 'style': None},
+            logging.WARNING: {'fg': 'blue', 'bg': 'default', 'style': None},
+            logging.ERROR: {'fg': 'red', 'bg': 'default', 'style': 'bold'},
+            logging.CRITICAL: {'fg': 'white', 'bg': 'red', 'style': 'bold'},
+        },
+    }
+
+    terminal_background: str = 'dark'
+
+    def set_terminal_background(self, background):
+        if background == 'auto':
+            self.terminal_background = ConsoleHelper.detect_background()
+        else:
+            self.terminal_background = background
+
+    def emit(self, record):
+        try:
+            message = self.format(record)
+            level_settings = self.colors[self.terminal_background].get(record.levelno, {})
+            ConsoleHelper.cprint(message, **level_settings)
+            self.flush()
+        except Exception:  # pylint: disable=broad-except
+            self.handleError(record)
+
+
 class LoggerHelper:
-    """
-    Helper class for setting up a logger
-    """
+    """Helper class for setting up a logger."""
 
     @staticmethod
-    def get_basic_logger(logger_name, level=logging.DEBUG):
+    def get_logger(logger_name: str, level: int = logging.DEBUG) -> logging.Logger:
         """Sets up a basic logger without any handler.
 
         Args:
-            logger_name (str): Logger name.
-            level (int): Severity threshold.
+            logger_name: Logger name.
+            level: Severity threshold.
 
         Returns:
-            logging.Logger: Created logger instance.
+            Created logger instance.
 
         """
+        logging.setLoggerClass(CustomLogger)
```

Comment:
Is this method really necessary? `logging.setLoggerClass(CustomLogger)` can be called just once when the module is loaded, it shouldn't be here.

Reply:
I initially didn't want to use the method and just use `logging.getLogger(__name__)` everywhere instead. Where would you put the `setLoggerClass` call to ensure that it gets called? For the use case of rebase-helper it could be called at the module level of logger.py but what about if one wanted to use only a part of the API (e.g. SpecFile class). We need to make sure that logger class gets properly set even in such case.

Reply:
That's a good point. Let me think about it.

Reply:
Well, what about `rebasehelper/__init__.py`?